### PR TITLE
Updating the included webserver to support searching

### PR DESCRIPTION
**Pull request by traetox**
This includes an updated golang webserver that holds an in-memory index and search API so that the documentation can perform some basic searching and link to anchor tags.

The in-memory index is NOT efficient, but it works and the data set is small.

### CODE REVIEW

File: serve.go [hunk 1]
```diff
@@ -1,10 +1,112 @@
...
+	sync.Mutex
+	http.Server
+	blink string //the base link anchor
+	dir   string
+	links map[string][]string
+}
+
+func main() {
+	flag.Parse()
+	if *lp == 0 || *lp > 0xffff {
```
Comment (djfritz) [inline serve.go:51]: This is somewhat unnecessary  as ListenAndServe() will fail when it attempts to bind.

**Comment by traetox** [comment]
Updated to lowercase everything so queries don't have to worry about case.

We also updated the index structure to have some more metadata that will make building a search interface better.

**Comment by traetox** [comment]
this: `curl -s --header "Content-Type: application/json"   --request POST   --data '{"value": "enumerated value JSON strict"}' http://localhost:3001/api/search`

now returns this:
```
{
  "links": [
    {
      "Page": "Search Extraction Modules",
      "Heading": "Universal Flags",
      "Link": "#!search/extractionmodules.md#Universal_Flags"
    },
    {
      "Page": "Search Processing Modules",
      "Heading": "Universal Flags",
      "Link": "#!search/processingmodules.md#Universal_Flags"
    },
    {
      "Page": "JSON",
      "Heading": "Empty Fields and the Strict Flag",
      "Link": "#!search/json/json.md#Empty_Fields_and_the_Strict_Flag"
    },
    {
      "Page": "JSON",
      "Heading": "Supported Options",
      "Link": "#!search/json/json.md#Supported_Options"
    }
  ]
}
```

File: search/dump/dump.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+## Dump
+
+The `dump` module is designed to treat resources as query-able data, meaning that you can inject a CSV or lookup table into the query pipeline and perform operations on the rows as if they are full fledged search entries.  Dump can be particularly useful when building dashboards and views that need to operate on data that has already been imported from some other static source or aggregated used scheduled queries.
```
Comment (djfritz) [inline search/dump/dump.md:3]: s/query-able/queryable/

File: search/dump/dump.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+### Supported Options
+
+* `-r`: The “-r” option requires an argument and specifies the name or UUID of a resource to extract data from.  A resource must be specified and the current user must have read access to that resource.
+* `-p`: The "-p" indicates that the dump module is injecting values along side other data and the pipeline should query the indexers.
+
+### Filtering and Column Inclusion
+
+The `dump` module will extract all columns from a CSV or lookup table resource if no columns are specified or if all specified columns have filters operations.  For example `dump -r hosts hostname` means only extract the hostname column, where `dump -r hosts hostname == "ad.example.com"` means extract all columns because every column specified has a filter operation attached.  The query `dump -r hosts hostname == "ad.example.com" IP` will only extract the `hostname` and `IP` columns.
```
Comment (djfritz) [inline search/dump/dump.md:14]: s/For example/For example,/

File: serve.go (hunk 1 above)
Comment (traetox) [reply serve.go:52]: i know, but i like throwing a super specific error.

File: search/dump/dump.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+Retrieve all columns from the `hosts` resource where the hostname is not empty and the owning org is "finance":
+```
+dump -r hosts hostname != "" org=="finance"
+```
+
+### Example Queries
+
+Dumps an entire resource into the pipeline and populates the table module:
```
Comment (djfritz) [inline search/dump/dump.md:47]: s/Dumps/Dump/

**Pull request merged**